### DISCUSSION TITLE
Does Spacy supports Relation Detection along with custom NER training?

### DISCUSSION THREAD

**Discussion by Muhammadusmanzafar** [explosion/spaCy#5309] [New Features & Project Ideas]
I am looking to train different sentences to extract NER and relation . NER is very well documented and I am able to achieve the results I was looking for in the form of entities. But I also want to detect Relations between those entities and want to train them along with NER .Does space supports this ?

**Comment by adrianeboyd** [comment]
No, spacy doesn't currently support this. It's likely that support will be added in the future, but there's no concrete timeframe right now.

**Comment by Tarun3679** [comment]
Hi, does it support now?

**Comment by svlandeg** [comment]
I'm afraid this is not currently on the planning to include in the core library. With the [pre-release of spaCy v3](https://github.com/explosion/spaCy/releases/tag/v3.0.0rc1) though, it should become much easier to implement your own [custom pipeline components](https://nightly.spacy.io/usage/processing-pipelines#trainable-components) with their [custom models](https://nightly.spacy.io/usage/layers-architectures). In fact, the documentation for this nightly release outlines the steps required to implement a relation detection algorithm: https://nightly.spacy.io/usage/layers-architectures#component-rel

I'll close this for now as there is no action item for us to follow up on, but wishing you good luck if you do decide to implement this ;-) And if you do - please also consider submitting it to the [spaCy universe](https://nightly.spacy.io/universe)!

Feel free to open a new issue if you run into specific problems!
